### PR TITLE
refactor: revamp event processing module unit test cases

### PR DESCRIPTION
**Pull request by SKannaniOS**
## Description
This PR refactors and modernizes the event processing unit tests by migrating from XCTest to the new Swift Testing framework. The changes focus on improving test organization, reliability, and maintainability by introducing new test structures and comprehensive mock utilities.

**Key Changes:**
- Migrated from XCTest to Swift Testing framework for event processing tests
- Reorganized test files into a dedicated `EventProcessingModuleTests` directory
- Enhanced mock storage with detailed event tracking and async wait utilities
- Improved test isolation and reliability with better async handling
- Added comprehensive test coverage for EventQueue, EventWriter, and EventUploader components

## Type of change
- [x] Code refactor/optimization
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

### Test Framework Migration
- **EventQueueTests**: Migrated from XCTest to Swift Testing with improved async event handling
- **EventWriterTests**: New comprehensive test suite for EventWriter component operations
- **EventUploaderTests**: Enhanced test coverage for upload scenarios including error handling and retry logic

### Test Organization
- Created new `EventProcessingModuleTests` directory replacing the old `EventQueue` folder
- Removed legacy test files: `EventUploadErrorTests.swift` and outdated `EventUploaderTests.swift`
- Consolidated test functionality into focused, well-structured test suites

### Mock Infrastructure Improvements
- **Enhanced MockStorage**: Added detailed event tracking with `currentBatchEventCount` and `totalEventCount`
- **Async Wait Utilities**: New methods like `waitForEvents()`, `waitForEventsContaining()`, and `waitForCurrentBatchEvents()`
- **Improved Test Isolation**: Better cleanup and state management between tests

### EventQueue Component Changes
- Made `start()` and `stop()` methods public for better testability
- Enhanced EventUploader with public `uploadBatch()` method for direct testing

### Test Coverage Enhancements
- **EventQueue**: Tests for single/multiple event processing, flush operations, anonymous ID changes
- **EventWriter**: Tests for event writing, rollover triggers, channel management
- **EventUploader**: Comprehensive error handling tests (400, 401, 404, 413, network errors)
- **Anonymous ID Extraction**: Detailed tests for various JSON formats and edge cases

### Modernized Test Utilities
- Streamlined SwiftTestMockProvider with improved mock URL session handling
- Removed redundant mock classes and consolidated functionality
- Enhanced JSON handling utilities for better test data management

## Checklist
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] I have added the necessary documentation (if appropriate).
- [x] I have ensured that my code follows the project's code style.
- [x] I have checked for potential performance impacts and optimized if necessary.
- [x] I have checked the code for security issues.
- [ ] I have updated the changelog (if required).

## How to test?
The refactored tests can be run using the standard Swift testing commands:
```
# Run specific event processing tests
swift test --filter EventQueueTests
swift test --filter EventWriterTests  
swift test --filter EventUploaderTests
```

**Test Scenarios Covered:**
1. **Event Processing**: Single and multiple event handling, different event types
2. **Batch Management**: Event accumulation, rollover triggers, anonymous ID changes
3. **Upload Error Handling**: Various HTTP error codes (400, 401, 404, 413), network failures, timeouts
4. **Channel Operations**: Proper channel closure, concurrent operations
5. **Storage Operations**: Event counting, batch tracking, cleanup procedures

**Expected Results:**
- All tests should pass with the new Swift Testing framework
- Test execution should be faster and more reliable due to improved async handling
- Better test isolation should reduce flaky test scenarios

## Breaking Changes
This is a refactoring change that doesn't introduce breaking changes to the public API. The changes are internal to the test suite and don't affect SDK functionality.

**Migration Notes for Contributors:**
- Test files have been reorganized into `EventProcessingModuleTests` directory
- New Swift Testing syntax should be used for future event processing tests
- Mock utilities have been enhanced with new async wait methods

## Maintainers Checklist
- [ ] The code has been reviewed.
- [ ] CI tests have passed.
- [ ] All necessary documentation has been updated.

## Additional Context
This refactoring is for revamping unit test cases in the event processing module. The migration to Swift Testing framework provides:

- **Better Async Support**: Native async/await support in test framework
- **Improved Reliability**: Better handling of concurrent operations and timing-sensitive tests
- **Enhanced Readability**: More expressive test syntax and better organization
- **Future-Proofing**: Alignment with modern Swift testing practices

The changes maintain backward compatibility while significantly improving test maintainability and reliability. The enhanced mock infrastructure will also benefit future test development across the SDK.

### CODE REVIEW

File: Tests/RudderStackAnalyticsTests/EventProcessingModuleTests/EventQueueTests.swift [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+        let result = await mockStorage.read()
+        #expect(result.dataItems.count > 0)
+        
+        let batchContent = result.dataItems.map { $0.batch }.joined()
+        #expect(batchContent.contains("event_1"))
+        #expect(batchContent.contains("event_2"))
+        #expect(batchContent.contains("event_3"))
+    }
+
+    @Test("given empty storage, when tracking events incrementally, then current batch counter accurately reflects event accumulation")
```
Comment (1abhishekpandey) [inline Tests/RudderStackAnalyticsTests/EventProcessingModuleTests/EventQueueTests.swift]: Regarding this: "when tracking events incrementally" - This is not super clear. Can you please improve it further?

File: Tests/RudderStackAnalyticsTests/EventProcessingModuleTests/EventUploaderTests.swift [hunk 2]
```diff
@@ -0,0 +1,364 @@
...
+        Task.detached {
+            await storage.removeAll()
+        }
+    }
+    
+    // MARK: - Anonymous ID Extraction Tests
+    
+    @Test("given batch with valid anonymousId, when extracting anonymousId, then correct id is returned")
+    func testExtractAnonymousIdFromValidBatch() async {
+        // Given
```
Comment (1abhishekpandey) [inline Tests/RudderStackAnalyticsTests/EventProcessingModuleTests/EventUploaderTests.swift]: Let's skip writing `given, when and then` as a comment.

**Review by 1abhishekpandey — APPROVED** [review]
